parteh/PRTAllometricCarbonMod.F90

```diff
@@ -0,0 +1,981 @@
+          ! At that point, update the actual states
+          ! --------------------------------------------------------------------------------
+          if( (totalC < calloc_abs_error) .and. (step_pass) )then
+
+             ierr           = 0
+             leaf_c_flux    = c_pool(leaf_c_id)   - leaf_c
+             fnrt_c_flux    = c_pool(fnrt_c_id)   - fnrt_c
+             sapw_c_flux    = c_pool(sapw_c_id)   - sapw_c
+             store_c_flux   = c_pool(store_c_id)  - store_c
+             struct_c_flux  = c_pool(struct_c_id) - struct_c
+             repro_c_flux   = c_pool(repro_c_id)  - repro_c
+             
+             ! Make an adjustment to flux partitions to make it match remaining c balance
+             flux_adj       = carbon_balance/(leaf_c_flux+fnrt_c_flux+sapw_c_flux + &
+                                              store_c_flux+struct_c_flux+repro_c_flux)
+
+             
+             leaf_c_flux    = leaf_c_flux*flux_adj
+             fnrt_c_flux    = fnrt_c_flux*flux_adj
+             sapw_c_flux    = sapw_c_flux*flux_adj
+             store_c_flux   = store_c_flux*flux_adj
+             struct_c_flux  = struct_c_flux*flux_adj
+             repro_c_flux   = repro_c_flux*flux_adj
+             
+             carbon_balance = carbon_balance - leaf_c_flux
+             leaf_c         = leaf_c + leaf_c_flux
+             
+             carbon_balance = carbon_balance - fnrt_c_flux
+             fnrt_c         = fnrt_c + fnrt_c_flux
+             
+             carbon_balance = carbon_balance - sapw_c_flux
+             sapw_c         = sapw_c + sapw_c_flux
+             
+             carbon_balance = carbon_balance - store_c_flux
+             store_c        = store_c + store_c_flux
+             
+             carbon_balance = carbon_balance - struct_c_flux
+             struct_c       = struct_c + struct_c_flux
+             
+             carbon_balance = carbon_balance - repro_c_flux
+             repro_c        = repro_c  + repro_c_flux
+             
+             dbh            = c_pool(dbh_id)
+
+             ! THESE HAVE TO BE SET OUTSIDE OF THIS ROUTINE
+             !!          cohort%seed_prod = cohort%seed_prod + brepro_flux / hlm_freq_day
+             !!          cohort%dhdt      = (h_sub-cohort%hite)/hlm_freq_day
+             !!          cohort%ddbhdt    = (dbh_sub-dbh_in)/hlm_freq_day
+             
+             if( abs(carbon_balance)>calloc_abs_error ) then
+                write(fates_log(),*) 'carbon conservation error while integrating pools'
+                write(fates_log(),*) 'along alometric curve'
+                write(fates_log(),*) 'carbon_balance = ',carbon_balance,totalC
+                write(fates_log(),*) 'exiting'
+                call endrun(msg=errMsg(sourcefile, __LINE__))
+             end if
+             
+          end if
+       end do
+    end if
+
+    ! Track the net allocations and transport from this routine
+
+    this%variables(leaf_c_id)%net_art(icd) = &
```

Comment:
Still a bit confused about why allocation is 'art'. Apologies if you addressed that already. Could it be net_alloc instead?

Reply:
Well,  the fluxes that this is tracking, is both the mass that is gained in the tissue, from daily carbon gain, or nutrient uptake etc (ie allocation).  It also tracks the net gains from retranslocation, which is typically the shuffling of carbon out of storage, or from dying leaves into the fresher leaves.  But I suppose, that is allocation as well?

Reply:
thus Allocation and Reactive Transport.

I suppose if anything, the "reactive" part of the ART is not really needed.   Is transport or retranslocation part of allocation...?